_health-care/_js/utils/validations.js

```diff
@@ -1,58 +1,207 @@
-  return /^\d{3}-\d{2}-\d{4}$/.test(value);
+  if (value !== null) {
+    return /^\d{3}-\d{2}-\d{4}$/.test(value);
+  }
+  return true;
 }
 
 function isValidDate(day, month, year) {
-  // Use the date class to see if the date parses back sanely as a
-  // validation check. Not sure this is a great idea...
-  const adjustedMonth = Number(month) - 1;  // JS Date object 0-indexes months. WTF.
-  const date = new Date(year, adjustedMonth, day);
-  return date.getDate() === Number(day) &&
-    date.getMonth() === adjustedMonth &&
-    date.getFullYear() === Number(year);
+  if (day !== null && month !== null && year !== null) {
+    // Use the date class to see if the date parses back sanely as a
+    // validation check. Not sure this is a great idea...
+    const adjustedMonth = Number(month) - 1;  // JS Date object 0-indexes months. WTF.
+    const date = new Date(year, adjustedMonth, day);
+    return date.getDate() === Number(day) &&
+      date.getMonth() === adjustedMonth &&
+      date.getFullYear() === Number(year);
+  }
+  return true;
 }
 
 function isValidName(value) {
-  return /^[a-zA-Z '\-]+$/.test(value);
+  if (value !== null) {
+    return /^[a-zA-Z '\-]+$/.test(value);
+  }
+  return true;
 }
 
 function isValidMonetaryValue(value) {
-  return /^\d+\.?\d*$/.test(value);
+  if (value !== null) {
+    return /^\d+\.?\d*$/.test(value);
+  }
+  return true;
 }
 
 // TODO: look into validation libraries (npm "validator")
 function isValidPhone(value) {
-  return /^\d{3}-\d{3}-\d{4}$/.test(value);
+  if (value !== null) {
+    return /^\d{3}-\d{3}-\d{4}$/.test(value);
+  }
+  return true;
 }
 
 function isValidEmail(value) {
-  // Comes from StackOverflow: http://stackoverflow.com/questions/46155/validate-email-address-in-javascript
-  return /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(value);
+  if (value !== null) {
+    // Comes from StackOverflow: http://stackoverflow.com/questions/46155/validate-email-address-in-javascript
+    return /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/.test(value);
+  }
+  return true;
 }
 
 // TODO:  1. what is a valid address?
 //        2. 6 arguments to a function is ugly...
 //        3. argument order is now based on form order... using
 function isValidAddress(street, city, country, state, zipcode) {
+  if (street !== null && city !== null && country !== null && state !== null && zipcode !== null) {
+    return true;
+  }
   // arbitraty use of field to keep linter happy until we answer #1
-  let n = 0;
-  if (street === '') n++;
-  if (city === '') n++;
-  if (country === '') n++;
-  if (state === '') n++;
-  if (zipcode === '') n++;
   return true;
 }
 
+function isValidNameAndGeneralInformation(data) {
+  return this.isNotBlank(data.fullName.first) &&
```

Comment:
The use of `this` here is weird.

It's probably working cause you're in a module and `this` is the module...but really, you should just be calling the function via name-lookup in lexical scoping. Drop all the `this.`?